dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -2664,24 +2674,29 @@ Spdp::SpdpTransport::write_i(const DCPS::RepoId& guid, WriteFlags flags)
     RTPS::log_message("(%P|%t) {transport_debug.log_messages} %C\n", hdr_.guidPrefix, true, message);
   }
 
-  send(flags);
+  send(flags, local_address);
 }
 
 void
-Spdp::SpdpTransport::send(WriteFlags flags)
+Spdp::SpdpTransport::send(WriteFlags flags, const ACE_INET_Addr& local_address)
 {
   DCPS::RcHandle<Spdp> outer = outer_.lock();
   if (!outer) return;
 
-  if ((flags & SEND_TO_LOCAL) && !outer->config_->rtps_relay_only()) {
+  if ((flags & SEND_MULTICAST) && !outer->config_->rtps_relay_only()) {
     typedef OPENDDS_SET(ACE_INET_Addr)::const_iterator iter_t;
     for (iter_t iter = send_addrs_.begin(); iter != send_addrs_.end(); ++iter) {
       send(*iter);
     }
   }
 
+  if (((flags & SEND_DIRECT) || !outer->config_->rtps_relay_only()) &&
```

Comment:
Why is this an `||`?  If the command is to send it "directly" and the configuration is "relay only," should that prevent this `send()`?